keps/sig-api-machinery/2885-server-side-unknown-field-validation/README.md

```diff
@@ -491,6 +520,23 @@ TypedValue](https://github.com/kubernetes/kubernetes/blob/9ff3b7e744b34c099c1405
 object has any unknown or duplicate fields.
 
 
+#### Kubectl Flag
+
+For beta, we will modify the kubectl `--validate` flag from being a bool flag to
+a string flag that accepts the following values:
+
+* `true` or `strict`: If server-side validation is enabled on the server it sends
```

Comment:
The conditional behavior based on the servers ability sounds confusing to me. Also, what do you do on a "client-dry-run"? i.e. `kubectl create --dry-run=client --validate`? Does that automatically fall back to client-side validation? How do you explain that? "Why are the errors looking so different"?

Reply:
Added a note that we will treat `--dry-run=client` as not being connected to any server at-all and thus will trigger a fallback to CS validation

Reply:
doesn't dry-run=client trigger local mode and prevent getting the schema from the server at all, and therefore do no validation?

Reply:
@apelisse do you know if there's some mechanism under the hood that using a cached schema or something. 

FWIW I can definitely see that validation _**is**_ being performed with `--dry-run=client` when running a simple experiment on a generic 1.23 cluster

Reply:
Yes, client dry-run still asks the server for the openapi. It has pros and cons ...